dist/transport.go

```diff
@@ -0,0 +1,186 @@
+package dist
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/transports"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+)
+
+//nolint (gochecknoinits)
+func init() {
+	transports.Register(Transport)
+}
+
+// nolint (gochecknoglobals)
+var Transport = distTransport{}
+
+type distTransport struct{}
+
+func (s distTransport) Name() string {
+	return "dist"
+}
+
+func splitReference(ref string) (fullname, server string, port int, err error) {
+	port = 8080
```

Comment:
This value should be hard-coded in _exactly one_ place, and in a top-level constant, not both here and in `NewOciRepo`.